### DISCUSSION TITLE
Using default MTU value warning when using client.mtu_size()

### DISCUSSION THREAD

**Discussion by ChrisM0815** [hbldh/bleak#1166] [Q&A] [answered]
* bleak version: 0.19.5
* Python version: 3.10.8
* Operating System: Linux 
* BlueZ version (`bluetoothctl -v`) in case of Linux: 5.55

### Description

When I call `client.mtu_size(self)` I´m getting the following warning on a Raspberry Pi:
`bleak/backends/bluezdbus/client.py:561: UserWarning: Using default MTU value. Call _acquire_mtu() or set _mtu_size first to avoid this warning.`

### What I Did
```python
import asyncio
import sys

from bleak import BleakScanner, BleakClient
from bleak.backends.winrt.client import logger as bleak_log

DEFAULT_MAC = "05:55:F5:73:C5:E7"
async def main(ble_address: str):
    print("Connecting")
    device = await BleakScanner.find_device_by_address(ble_address, timeout=20.0)
 
    async with BleakClient(device) as client:
        print(client.mtu_size)
        
if __name__ == "__main__":
    asyncio.run(main(sys.argv[1] if len(sys.argv) == 2 else DEFAULT_MAC))
```
I´m using this Script to connect to a ble device and print out the MTU. On Linux I ´m getting the above mentioned warning. I was not able to call the `client._acquire_mtu()` method. Is there a way to get a higher MTU on Linux or at least prevent this warning from printing?
Thanks

**Comment by dlech** [comment, marked as answer]
If possible, you should use [BleakGATTCharacteristic.max_write_without_response_size](https://bleak.readthedocs.io/en/latest/api/index.html#bleak.backends.characteristic.BleakGATTCharacteristic.max_write_without_response_size) instead of `BleakClient.mtu_size`. 

To avoid the warning, call `client._backend._mtu_size = EXPECTED_MTU` before reading `client.mtu_size`.

**Comment by EdwinFairchild** [comment]
is Apple macOS able to get MTU from `client.mtu_size` ? I have no way of testing.

**Reply by dlech** [reply]
It is recommended to use https://bleak.readthedocs.io/en/latest/api/index.html#bleak.backends.characteristic.BleakGATTCharacteristic.max_write_without_response_size in new code instead. Both work on macOS.

**Reply by EdwinFairchild** [reply]
yeah I saw that , but I am trying to get MTU size of a server that will request  a smaller one and the characteristics I am using are not write without response, I act based on the response.

**Reply by EdwinFairchild** [reply]
Also saw the work around in mtu_size.py but inspite of being on Ubuntu the if statement fails, I used other methods to test if I am linux and I am setting a smaller MTU evne though I cannot really get the actual MTU size on linux. On windows it works.

**Reply by dlech** [reply]
It doesn't matter what the characteristic supports it will still return the same value. `max_write_without_response_size` is still the best way to get the value cross-platform otherwise you have to provide your own implementation for Linux. If you really need MTU, you can just add 3 to this value.
